pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -554,15 +554,24 @@ func (c *Cluster) startMon(m *monConfig, hostname string) error {
 	}
 
 	d := c.makeDeployment(m, hostname)
-	logger.Debugf("Starting mon: %+v", d.Name)
+	logger.Debugf("Starting mon deployment: %+v", d)
 	_, err := c.context.Clientset.Extensions().Deployments(c.Namespace).Create(d)
 	if err != nil {
 		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create mon %s. %+v", m.ResourceName, err)
+			return fmt.Errorf("failed to create mon deployment %s. %+v", m.ResourceName, err)
+		}
+		logger.Infof("deployment for mon %s already exists. updating if needed", m.ResourceName)
+		if err := k8sutil.UpdateDeploymentAndWait(c.context, d, c.Namespace); err != nil {
+			return fmt.Errorf("failed to update mon deployment %s. %+v", m.ResourceName, err)
 		}
-		logger.Debugf("mon %s already exists", m.ResourceName)
 	}
 
+	// TODO: wait to return until the mon we are starting up reports that it is healthy ...
```

Comment:
How about if we check for mon quorum and doc updates in separate PRs? We can keep this one scoped to updating the deployments.

To check for mon quorum, we could hopefully re-use the method `waitForQuorumWithMons()` and just wait for quorum before continuing. I've seen issues in the integration test  because we don't wait, so this will be a very positive thing. If something else is going on with the mons, that method should time out and the operator would continue in that case.